File: SDKs/microsoft.com/cognitiveservices-Training/2.1/go/pkg/models/operations/getiteration.go
package operations

import (
	"openapi/pkg/models/shared"
)

type GetIterationPathParams struct {
	IterationID string `pathParam:"style=simple,explode=false,name=iterationId"`
	ProjectID   string `pathParam:"style=simple,explode=false,name=projectId"`
}

type GetIterationHeaders struct {
	TrainingKey string `header:"style=simple,explode=false,name=Training-Key"`
}

type GetIterationRequest struct {
	PathParams GetIterationPathParams
	Headers    GetIterationHeaders
}

type GetIterationResponse struct {
	Body        []byte
	ContentType string
	Iteration   *shared.Iteration
	StatusCode  int64
}

File: SDKs/microsoft.com/cognitiveservices-Training/2.0/typescript/src/sdk/models/operations/getiteration.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";


export class GetIterationPathParams extends SpeakeasyBase {
  @Metadata({ data: "pathParam, style=simple;explode=false;name=iterationId" })
  iterationId: string;

  @Metadata({ data: "pathParam, style=simple;explode=false;name=projectId" })
  projectId: string;
}


export class GetIterationHeaders extends SpeakeasyBase {
  @Metadata({ data: "header, style=simple;explode=false;name=Training-Key" })
  trainingKey: string;
}


export class GetIterationRequest extends SpeakeasyBase {
  @Metadata()
  pathParams: GetIterationPathParams;

  @Metadata()
  headers: GetIterationHeaders;
}


export class GetIterationResponse extends SpeakeasyBase {
  @Metadata()
  body?: Uint8Array;

  @Metadata()
  contentType: string;

  @Metadata()
  iteration?: shared.Iteration;

  @Metadata()
  statusCode: number;
}
